networkx/algorithms/tests/test_dag.py

```diff
@@ -121,6 +121,47 @@ def test_descendants(self):
         assert_equal(descendants(G, 3), set())
         assert_raises(nx.NetworkXError, descendants, G, 8)
 
+    def test_transitive_closure(self):
+        G = nx.DiGraph([(1, 2), (2, 3), (3, 4)])
+        transitive_closure = nx.algorithms.dag.transitive_closure
+        assert_equal(transitive_closure(G).edges(), [(1, 2), (1, 3),
+                                                     (1, 4), (2, 3), (2, 4),
+                                                     (3, 4)])
+        G = nx.DiGraph([(1, 2), (2, 3), (2, 4)])
+        assert_equal(transitive_closure(G).edges(), [(1, 2), (1, 3),
+                                                     (1, 4), (2, 3), (2, 4)])
+        G = nx.Graph([(1, 2), (2, 3), (3, 4)])
+        assert_raises(nx.NetworkXNotImplemented, transitive_closure, G)
+
+    def test_antichains(self):
+        antichains = nx.algorithms.dag.antichains
+        G = nx.DiGraph([(1, 2), (2, 3), (3, 4)])
+        assert_equal(list(antichains(G)), [[], [4], [3], [2], [1]])
```

Comment:
@ysitu, I think that the output of `transitive_clousure` is deterministic (ie. regardless of the order of the iteration `for v in G`). However the output order of `antichains` can change if the topological sort of the transitive clousure is not unique (ie. there is no Hamiltonian path in `TC`). Thus the test are indeed fragile.